src/App.vue

```diff
@@ -1,6 +1,23 @@
+<script setup lang="ts">
+import { provide, shallowRef } from "vue"
+import router from "./application/router";
+import { layouts } from "./presentation/layouts";
+
+const layout = shallowRef("div");
+
+router.afterEach((to) => {
+  layout.value = layouts[to.meta.layout] || "div";
+});
+
+provide("app:layout", layout);
+
+</script>
+
 <template>
   <div id="app">
-    <router-view></router-view>
+    <component :is="layout || 'div'">
```

Comment:
maybe it would be cleaner to move this part to the `presentation/layouts/Layout.vue` and get rid of `layouts/index.ts`?